### ISSUE TITLE
Idea/Suggestion - Generate Code - HTTP File with Variables 

### ISSUE THREAD

**Issue by chrismalek** [httpyac/httpyac.github.io#54]
Again I want to praise this plugin. I use it all day every day and is amazing!!!

One idea that you may consider.

When you "Generate code" it would be really cool/helpful if there was another option to generate an HTTP file with all the variables substituted that I can execute. It seems the HTTP syntax cannot be executed by HTTPYac.

Here is my use case.

On my local machine, I often have all sorts of .env files and different HTTP files. However, my local machine often does not have connections to the target HTTP servers due to network restrictions at different clients. I often go into bastion hosts/vms that have vscode installed but don't have all my .env files and other config that httpyac will pull in from my local machine. I often use the "Generate Code" --> HTTP. However, the HTTP syntax does not seem to be able to be executed directly by HTTPYac and requires some manipulation (adding https:// and moving the hostname header value to the URL). So I am thinking here if there was another template that was similar to the http code generation but that puts in the full URL. This way I can just copy and past the http file and put it in the bastion host and run it.  I have tried the curl and it sometimes does not work in httpyac.  At the time of writing I don't recall the issue I was having but I will loop back.

**Comment by AnWeber** [comment]
@chrismalek I solve the problem by using SSH port forwarding. Instead of installing/running `httpyac` directly on the server, I enable SSH port forwarding and configure a socks proxy and use my local setup directly.

```sh
ssh -D 1111 user@jumphost
```

```http
# @proxy socks://127.0.0.1:1111
# @noRejectUnauthorized
GET https://securedserver:6443/openapi/v2
```
But to the actual question. What advantage over `curl` would it have to run `httpyac` on the server if there is no more variable substitution?

**Comment by chrismalek** [comment]
Often times port forwarding is not possible.  I often have to bastion into one machine then use another bastion machine to get access. This is RDP into RDP and I don't have control over these machines. Windows port forwarding not as easy. 

I did some more testing. When I generate a curl command locally that I want to then execute in httpyac on some other machine the first generation generates something like this:

```sh
curl --request POST --url 'https://httpbin.org/anything'  --header 'Content-Type: text/xml'  --header 'From: blahbah'  --header 'OperationName: some-service'  --data '<?xml version="1.0" encoding="utf-8"?>
<request>
    <foo>bar</foo>
</request>'

```


When I put that in another VSCODE / HTTPYAC HTTP file and try to execute it, HTTPYac throws an error 

![image](https://user-images.githubusercontent.com/9403/184429444-cb5717f2-dc4a-4555-8110-ef713770fd41.png)


![image](https://user-images.githubusercontent.com/9403/184429551-5f7f9081-a0c2-490b-ba1d-d65ed171b53e.png)


So this is NOT "portable." This is why I have been falling back on http because it is less manual manipulation to get it to work. I remove hostname and move it to the URL. It is not the end of the world.  I am a heavy user so I am just looking for some efficiencies.  😃  

If the curl would work as generated inside vscode at least it would be helpful. I can't tell if httpyac is interpreting the curl command incorrectly or if something is missing (or I am making a dumb mistake). 

I have also had clients where I give them that curl command and it will not work. I think the line continuation character might be the culprit some times across different operating systems. Sorry I don't have all the detail here of all the different errors I could get.

**Comment by AnWeber** [comment]
By using `curl` I did mean to use it in cli. I did not implement `curl` parsing, as I have not understood the purpose. The parsing assumes that the first line is the url if no request method is detected (GET, POST, ...). Due to the variables, it is possible to write the URL in too many different ways to still filter out the false positives.
I can include an additional option that simply transfers all external variables to the file as well (except `process.env`). It only gets difficult with `# @import` and request body import (`< ./payload.json`).

**Comment by chrismalek** [comment]
I must be confusing the curl parsing with another plugin.  No wonder why is was not working. Lol.

**Comment by AnWeber** [comment]
`vscode-restclient` does support incomplete curl parsing. But this feature was not worth implementing for me so far. 

I have now created a way to merge all executed Http Regions and variables together in one file. Maybe this will help you. 
<img width="889" alt="image" src="https://user-images.githubusercontent.com/13434739/184552286-d2cb49a1-49fb-40c4-9d74-73fa64de069c.png">

**Comment by AnWeber** [comment]
published with v5.6.2
